### ISSUE TITLE
devnet returns HTTP-500 when there is no param in the RPC call

### ISSUE THREAD

**Issue by gregoryguillou** [NethermindEth/caigo#66]
When running Caigo with devnet with calls that do not have parameters, like `starknet_protocolVersion`, the JSON RPC implementation of go-ethereum, send the following payload:

```json
{
    "id": 1,
    "jsonrpc": "2.0",
    "method": "starknet_protocolVersion"
}
```

That makes sense since there is no parameters, however, devnet and opposite to pathfinder fails with HTTP-500. We need to get the following payload:

```json
{
    "id": 1,
    "jsonrpc": "2.0",
    "method": "starknet_protocolVersion",
    "params": []
}
```

**Comment by gregoryguillou** [comment]
I have workaround the issue which I think is an issue with devnet rather than us by adding `[]interface{}{}...` in the RPC call which has as an effect to add the `"params": []` in the RPC call. It would have to be added to every single call.

Once fixed everywhere, the problem should be gone. #54 will do the job. Except for that, I have also seen some methods are not implemented like `starknet_syncing`.

**Comment by gregoryguillou** [comment]
The fix is included in #65 and will be covered by the addition of tests on devnet as described in #54. I will close that issue.
